types/uint.go

```diff
@@ -0,0 +1,177 @@
+package types
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/pkg/errors"
+)
+
+// Unt wraps integer with 256 bit range bound
+// Checks overflow, underflow and division by zero
+// Exists in range from 0 to 2^256-1
+type Uint struct {
+	i *big.Int
+}
+
+// NewUintFromBigUint constructs Uint from big.Uint
+func NewUintFromBigInt(i *big.Int) Uint {
+	u, err := checkNewUint(i)
+	if err != nil {
+		panic(fmt.Errorf("overflow: %s", err))
+	}
+	return u
+}
+
+// NewUint constructs Uint from int64
+func NewUint(n uint64) Uint {
+	i := new(big.Int)
+	i.SetUint64(n)
+	return NewUintFromBigInt(i)
+}
+
+// NewUintFromString constructs Uint from string
+func NewUintFromString(s string) Uint {
+	u, err := ParseUint(s)
+	if err != nil {
+		panic(err)
+	}
+	return u
+}
+
+// ZeroUint returns unsigned zero.
+func ZeroUint() Uint { return NewUint(0) }
+
+// OneUint returns Uint value with one
+func OneUint() Uint { return NewUint(1) }
+
+// Uint64 converts Uint to uint64
+// Panics if the value is out of range
+func (u Uint) Uint64() uint64 {
+	if !u.i.IsUint64() {
+		panic("Uint64() out of bound")
+	}
+	return u.i.Uint64()
+}
+
+// IsZero returns 1 if the uint equals to 0.
+func (u Uint) IsZero() bool { return u.Equal(ZeroUint()) }
+
+// Equal compares two Uints
+func (u Uint) Equal(u2 Uint) bool { return equal(u.i, u2.i) }
+
+// GT returns true if first Uint is greater than second
+func (u Uint) GT(u2 Uint) bool {
+	return gt(u.i, u2.i)
+}
+
+// GTE returns true if first Uint is greater than second
+func (u Uint) GTE(u2 Uint) bool { return u.GT(u2) || u.Equal(u2) }
```

Comment:
can we be consistent with one-lining functions? Either do them or don't for these (I prefer we don't).

Reply:
? What's the matter with inline functions? This one is pretty dumb...

Reply:
It's just inconsistent. Some are one-liners and some aren't' (e.g. `GT` vs GTE).

Reply:
fixed, thanks